### PR TITLE
Added custom Cypress command to create an article.

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Testing Infrastructure

## Description

A custom Cypress command for creating an article in end to end (E2E) tests. This is required to write E2E tests for #11983.

## Related Tickets & Documents

#11983

## QA Instructions, Screenshots, Recordings

N/A This is testing infrastructure.

### UI accessibility concerns?

N/A This is testing infrastructure.
## Added tests?

- [ ] Yes
- [x] No, and this is why: This is testing infrastructure.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

Animated GIF saying "New blog post!"

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -140,3 +140,63 @@ Cypress.Commands.add(
...
+        submitting: false,
+        editing: false,
+        mainImage: null,
+        organizations,
+        organizationId,
+        edited: true,
+        updatedAt: null,
+        version: editorVersion,
+        siteLogo:
+          '<a href="/" class="site-logo" aria-label="DEV(local) Community Home">\n    <span class="truncate-at-2">\n      DEV(local) Community\n    </span>\n</a>\n',
```
Comment (nickytonline) [inline cypress/support/commands.js:196]: When I inspected the Network tab to create this custom command, the request sent the site's logo is part of the new article creation. Any reason this is required? cc: @rhymes

File: cypress/support/commands.js [hunk 2]
```diff
@@ -140,3 +140,63 @@ Cypress.Commands.add(
...
+        helpFor: 'article_body_markdown',
```
Comment (nickytonline) [inline cypress/support/commands.js:197]: Not sure why the help position and which piece of help we're on is relevant to article creation either. Is this something to do with preview mode?

File: cypress/support/commands.js [hunk 3]
```diff
@@ -140,3 +140,63 @@ Cypress.Commands.add(
     );
   },
 );
+
```
Comment (nickytonline) [inline cypress/support/commands.js:143]: For now, we don't have a lot of custom commands, so I'm leaving them directly in commands.js. If we start to see this file become a little unwieldly, we'll revisit and most likely start importing custom commands. For now, no [YAGNI](https://martinfowler.com/bliki/Yagni.html).

File: cypress/support/commands.js (hunk 1 above)
Comment (rhymes) [reply cypress/support/commands.js:196]: I'm not very familiar with the Article's editor to be honest. That's weird. It looks like it's HTML embedded in a `data-` attribute 😕 

https://github.com/forem/forem/blob/11da7ddb9f09d55c35bddb9ecf866e18942adaa5/app/views/articles/_v2_form.html.erb#L43

It doesn't seem to be fetched in JS though. I wonder if it's a remnant of something older. It comes from https://github.com/forem/forem/pull/10840/files#diff-6f417f63e88e9b7e4b28f32d77d1d6ed33fd8d0500c7494e29fdc31bfc7b482bR20

File: docs/tests/e2e-tests.md [hunk 4]
```diff
@@ -108,6 +108,42 @@ Pro as there were issues integrating the cypress-rails gem with the
...
+```javascript
+cy.createArticle({
+  title: 'Test Article',
+  tags: ['beginner', 'discuss'], // tags are optional
+  content: 'This is a test article',
+});
+```
+
+If you want to do something with the article creation's response, you can call
+it like so.
```
Comment (rhymes) [inline docs/tests/e2e-tests.md:135]: cool!

File: cypress/support/commands.js (hunk 1 above)
Comment (nickytonline) [reply cypress/support/commands.js:196]: I removed `siteLogo`, `helpFor` and `helpPosition` and article creation still works. Regardless, still odd that it's all being sent.

**Comment by nickytonline** [comment]
I'm going to close this @rhymes. Although it does create an article, [given that Elastic Search runs after commit](https://docs.forem.com/backend/elasticsearch/) via a `after_commit` callback, this will be of no use to my E2E tests as tests run in a transaction that rollback, so Elastic Search will never kick in. I'm going to opt for seeding the E2E DB with a few articles to test this instead.

**Pull request closed without merging**